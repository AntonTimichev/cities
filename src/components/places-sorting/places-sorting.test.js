import React from "react";
import renderer from "react-test-renderer";

import PlacesSorting from "./places-sorting.jsx";

const options = {
  '0': `Popular`,
  '1': `Price: low to high`,
  '2': `Price: high to low`,
  '3': `Top rated first`
};

it(`PlacesSorting is rendered correctly`, () => {
  const tree = renderer.create(<PlacesSorting
    currentKey={-1}
    currentOption={`to Top`}
    options={options}
    isOpen={false}
    onToggleItemClick={jest.fn()}
    onSortingItemCLick={jest.fn()}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
