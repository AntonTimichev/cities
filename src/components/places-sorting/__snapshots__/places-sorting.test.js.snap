// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesSorting is rendered correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    to Top
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
    onClick={[Function]}
  >
    <li
      className="places__option"
      data-key="0"
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      data-key="1"
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      data-key="2"
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      data-key="3"
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
